test/errors-spec.js

```diff
@@ -0,0 +1,345 @@
+                            return {
+                                pass: false,
+                                message: "'" + actual + "'" +
+                                    " parsed with error '" + msg.slice(19) +
+                                    "' but expected '" + expected + "'"
+                            };
+                        } else {
+                            return {
+                                pass: false,
+                                message: "'" + actual + "'" +
+                                    " caused error '" + msg +
+                                    "' but expected '" + exp + "'"
+                            };
+                        }
+                    }
+                }
+            };
+        }
+    });
+});
+
+describe("Parser:", function() {
+
+    describe("#handleInfixNodes", function() {
+        // TODO: The position information here is broken, should be fixed.
+        it("rejects repeated infix operators", function() {
+            expect("1\\over 2\\over 3").toFailWithParseError(
+                   "only one infix operator per group at position -1: " +
+                   "1\\over 2\\over ");
+        });
+        it("rejects conflicting infix operators", function() {
+            expect("1\\over 2\\choose 3").toFailWithParseError(
+                   "only one infix operator per group at position -1: " +
+                   "1\\over 2\\choos");
+        });
+    });
+
+    describe("#handleSupSubscript", function() {
+        it("rejects ^ at end of group", function() {
+            expect("{1^}").toFailWithParseError(
+                   "Expected group after '^' at position 3: {1^̲}");
+        });
+        it("rejects _ at end of input", function() {
+            expect("1_").toFailWithParseError(
+                   "Expected group after '_' at position 2: 1_̲");
+        });
+        it("rejects \\sqrt as argument to ^", function() {
+            expect("1^\\sqrt{2}").toFailWithParseError(
+                   "Got function '\\sqrt' with no arguments as superscript" +
+                   " at position 2: 1^̲\\sqrt{2}");
+        });
+    });
+
+    describe("#parseAtom", function() {
+        // TODO: The positions in the following error messages appear to be
+        // off by one, i.e. they should be one character later.
+        it("rejects \\limits without operator", function() {
+            expect("\\alpha\\limits\\omega").toFailWithParseError(
+                   "Limit controls must follow a math operator" +
+                   " at position 6: \\alpha̲\\limits\\omega");
+        });
+        it("rejects \\limits at the beginning of the input", function() {
+            expect("\\limits\\omega").toFailWithParseError(
+                   "Limit controls must follow a math operator" +
+                   " at position 0: ̲\\limits\\omega");
+        });
+        it("rejects double superscripts", function() {
```

Comment:
To me, “ambiguous” is less specific than “double”. And since the error message literally mentions double superscripts, I see no harm but great benefit in sticking to the wording of the error message I'm testing here.